### PR TITLE
Added dependabot config for docker images and github actions

### PR DESCRIPTION
**Pull request by patcon**
Re-ticketed from https://github.com/GCTC-NTGC/gc-digital-talent/pull/1855#issuecomment-1020629661

This will use dependabot PRs to update external github actions, like here:

https://github.com/GCTC-NTGC/gc-digital-talent/blob/e24472cb3fbabc03ce6c470e8a7efe40a1b3aa63/.github/workflows/bundlewatch.yml#L39

and docker images like here:

https://github.com/patcon/gc-digital-talent/blob/39c1ad507912c40e86a9817259bbe5b35ca79cea/api/Dockerfile#L1

### CODE REVIEW

File: .github/dependabot.yml
Comment (patcon) [inline .github/dependabot.yml:10]: seemed redundant?

File: .github/dependabot.yml
Comment (patcon) [inline .github/dependabot.yml:53]: stuck with weekly

File: .github/dependabot.yml
Comment (patcon) [inline .github/dependabot.yml:54]: unlike npm, updates so rare that left unlimited

File: .github/dependabot.yml
Comment (petertgiles) [reply .github/dependabot.yml:10]: Lol, I guess they are pretty low-value comments. 😁

File: .github/dependabot.yml
Comment (petertgiles) [reply .github/dependabot.yml:53]: Yeah, lets keep it on the same cadence for simplicity.

File: .github/dependabot.yml
Comment (petertgiles) [reply .github/dependabot.yml:54]: Makes sense.

**Comment by petertgiles** [comment]
I'm pretty new to github actions but it makes sense that we would want to keep on top of any updates there.  I'm not sure about the docker images though.  I feel like the ones that aren't already set to "latest" are pinned to a specific version for a reason.  Am I missing something?

**Comment by patcon** [comment]
> I feel like the ones that aren't already set to "latest" are pinned to a specific version for a reason.

I'll admit I was laying groundwork to eventually open a PR removing our use of "latest" tags, for which pinning becomes manageable with dependabot. The value of these things is perhaps only surfaced once we start running e2e tests, where we can have confidence that even updates in docker image versions result in our important user stories still working as expected :)

As for "already pinned" versions, dependabot allows setting ignore patterns hardcoded in the config file itself, but I've found dependabot's in-PR commands have the best flow for ignoring stuff (and smartly re-enable if the `FROM foo:bar` tag is changed):

> - `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)

**Comment by petertgiles** [comment]
> I was laying groundwork to eventually open a PR removing our use of "latest" tags, for which pinning becomes manageable with dependabot

Ah, I get it.  Then we control when the "latest" images get updated and can review.  We would just need to be careful on the first round of updates to "repin" certain image versions.

I feel like I'm taking an online docker course!  👨‍🎓

**Comment by patcon** [comment]
> I feel like I'm taking an online docker course!

Pls do lemme know if it's TMI or docker overload :) Not my intention

> careful on the first round of updates to "repin" certain image versions.

Yeah, I think so :)

So if newest `foo:2.0.1` exists, but we only want the most recent 1.x series, we could use `foo:1.2.3` in dockerfile (the most recent 1.x). Because we're specific with the tag, dependabot should immediately open a PR for `foo:2.0.1` (assuming that's newest, but we can comment `@dependabot ignore this major version` and it will go away for 2.x, and only prompt us for updates if `foo:1.2.4` or `foo:1.3.0` comes out. (Or we could wholesale ignore the `foo` image updates with that command i mentioned earlier: `@dependabot ignore this dependency`)

Sorry if that's verbose, just figured was clearer with example

**Comment by yonikid15** [comment]
Very cool stuff. I'm learning a ton as well. Didn't know you can add dependabot config for docker images.

**Comment by patcon** [comment]
Thanks yoni!

@all Shall I remove any part of this for now though, and treat it separately? I don't want to hold this up if there's any hesitation

**Review by petertgiles — APPROVED** [review]
Thanks Pat!  Sorry for the hesitation, I just wanted to check in with the team before merging this since it wasn't associated with an existing issue on the board.  We're all good with it.

**Pull request merged**
